### PR TITLE
tools/notify: Reorder/cleanup maintainer lists

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/repo/notify.py [hunk 1]
```diff
@@ -34,43 +34,42 @@
...
+    'daixiang0': 'U020CJG6UU8',
+    'silverstar194': 'U03LNPC8JN9',
     'tyxia': 'U023U1ZN9SP',
 }
 
 # Only notify API reviewers who aren't maintainers.
 # Maintainers are already notified of pending PRs.
 API_REVIEWERS = {
     'markdroth': 'UMN8K55A6',
-    'adisuissa': 'UT17EMMTP',
```
Comment (nezdolik) [inline tools/repo/notify.py:73]: do we now have only one person as api reviewer?

**Review by nezdolik — APPROVED** [review]
lgtm, do we now have only one person as api reviewer?

File: tools/repo/notify.py (hunk 1 above)
Comment (phlax) [reply tools/repo/notify.py:73]: i was going on the comment - reasonably confident that last time i refactored the related code it respected existing way - so my reading is that there is no need to add in API_REVIEWERS

Comment (adisuissa) [reply tools/repo/notify.py:73]: I'm not an expert on the tooling, so I've got 2 questions:
1. will this impact the the PR notifier, and if so, how?
2. will this impact the auto-assigner of the api-reviewers?

Comment (phlax) [reply tools/repo/notify.py:73]: i believe the answer is no to both questions - but i can track through source and check

Comment (phlax) [reply tools/repo/notify.py:73]: wrt the 2nd question - totally different system https://github.com/envoyproxy/envoy/blob/b483d296738516839ec09ea3056f9c90be362080/repokitteh.star#L11-L20

Comment (phlax) [reply tools/repo/notify.py:73]: wrt 1st - reading source again im pretty confident it has no effect - its only used to notify personally and all maintainers are notified whenever they are assigned anyway (as per comment)

lmk if for some reason this is not correct

**Pull request merged**